testing/e2e/suites/defaultdomain.go

```diff
@@ -0,0 +1,115 @@
+package suites
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/aks-app-routing-operator/api/v1alpha1"
+	"github.com/Azure/aks-app-routing-operator/pkg/util"
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/infra"
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/logger"
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/manifests"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func defaultDomainTests(in infra.Provisioned) []test {
+	return []test{
+		{
+			name: "default domain certificate crd rejection tests",
+			cfgs: builderFromInfra(in).
+				withOsm(in, false, true).
+				withVersions(manifests.OperatorVersionLatest).
+				withZones(manifests.AllDnsZoneCounts, manifests.AllDnsZoneCounts).
+				build(),
+			run: func(ctx context.Context, config *rest.Config, operator manifests.OperatorConfig) error {
+				lgr := logger.FromContext(ctx)
+				lgr.Info("Running default domain certificate crd rejection tests")
+
+				scheme := runtime.NewScheme()
+				v1alpha1.AddToScheme(scheme)
+				cl, err := client.New(config, client.Options{Scheme: scheme})
+				if err != nil {
+					return fmt.Errorf("creating client: %w", err)
+				}
+
+				cases := []struct {
```

Comment:
can you add validation for the specific error we expect

Reply:
done